### ISSUE TITLE
Is it possible to use overloads?

### ISSUE THREAD

**Issue by divan** [atn832/cel-dart#1]
Hi, I'm trying to understand if it's possible to use this library with existing expressions from cel-go. In cel-go it's possible to declare custom functions (overloads) and use in expressions. README says that this project doesn't support passing custom environment options. Does it also mean that it's not possible to specify overloads?
If so, any plans to implement it?

**Comment by atn832** [comment]
You're right. So far the only `Environment` one can use is https://github.com/atn832/cel-dart/blob/cc76e33ddb3b19dfcafd9facae197a11c8d78a7a/lib/src/cel/environment.dart#L20-L28

And indeed, it also means you can't specify overloads yet.

What specific expressions from cel-go are you trying to run?

There seems to be some demand for it. Perhaps a contributor or myself could build it. See https://github.com/atn832/fake_firebase_security_rules/issues/2.

**Comment by divan** [comment]
@atn832 thanks for quick answer. Well, I'm yet exploring if CEL fits my needs or not. My use case is not related to networking, it's pretty custom stuff, so I probably need some custom methods/functions. Like one of the stuff I was implementing in cel-go veriant is "trimmed mean" function. Basically find min and max values in the int list, drop them, and calculate mean.

As I said, CEL might not be a good fit for my case, though I really like philosophy of it and approach, but I'm looking for something that will work both for Go and Dart, hence the question.
